configure.ac

```diff
@@ -151,6 +151,19 @@ AC_SUBST([git_commit_hash], "$GIT_COMMIT_HASH")
 AC_SUBST([DEV_VERSION_SUFFIX])
 AC_SUBST([dist_version], "$DIST_VERSION")
 
+AC_ARG_WITH([mono-framework],
+  AC_HELP_STRING([--with-mono-framework="MONO_FRAMEWORK"],
+  [Set where to look for the mono framework (macOS) [default=/Library/Frameworks/Mono.framework/Versions/Current]]),
+  with_mono_framework=$enableval,
+  with_mono_framework=/Library/Frameworks/Mono.framework/Versions/Current
```

Comment:
Thing is, if `WITH_MONO_FRAMEWORK` variable is only used in the Mac, then it should have a better mac-related name IMHO, something like MONO_FRAMEWORK_MAC_PATH?

And maybe move all the AC_ARG_WITH call inside a Darwin-if test.